R/generateSettings.R

```diff
@@ -40,78 +39,80 @@ generateSettings <- function(standard="None", chart="eDish", partial=FALSE, part
+  # Split on -- for multi-level handling 
   hierarchical_metadata <- str_split(metadata$text_key, "--") 
   
-  standard_low <- tolower(standard)
+  settings<-list(
+    id_col = NULL,
+    value_col = NULL,
+    measure_col = NULL,
+    normal_col_low = NULL,
+    normal_col_high = NULL,
+    studyday_col=NULL,
+    visit_col = NULL,
+    visitn_col = NULL,
+    filters = NULL,
+    group_cols = NULL,
+    measure_values = list(alt = NULL,
+                          ast = NULL,
+                          tb = NULL,
+                          alp = NULL),
+    baseline = list(value_col=NULL,
+                    values=list()),
+    analysisFlag = list(value_col=NULL,
+                        values=list()),
+    
+    x_options = c("LT", "AST", "ALP"),
+    y_options = c("TB", "ALP"),
+    visit_window = 30,
+    r_ratio_filter = TRUE,
+    r_ratio_cut = 0,
+    showTitle = TRUE,
+    warningText = "Caution: This interactive graphic is not validated. Any clinical recommendations based on this tool should be confirmed using your organizations standard operating procedures."
+  )
   
-  if (standard == "None") {
-    standard_low <- "default"
-  }
+  potential_settings <- settings
   
-  # Build empty settings list
-  potential_settings <- list()
-  for ( i in 1:length(hierarchical_metadata) ) {
+  standard_low <- tolower(standard)
+  
+  if (standard_low == "adam" | standard_low == "sdtm") {
     
-    # Handle settings with one level
-    if (length(hierarchical_metadata[[i]]) == 1) {
-
-      #Handle Nulls
-       if (is.null(parse_expr(metadata[[standard_low]][i]))) {
-         potential_settings[metadata$text_key[i]] = list(parse_expr(metadata[[standard_low]][i]))
-         
-        # Handle C()  - not implemented for level two yet
-       } else if (typeof(parse_expr(metadata[[standard_low]][i])) == "language") {
-         potential_settings[[metadata$text_key[i]]] = as.character(parse_expr(metadata[[standard_low]][i]))[-1]  #Need the -1 to remove the unwelcome "c" that comes with this method
-         
-       } else {
-        potential_settings[metadata$text_key[[i]]] = parse_expr( metadata[[standard_low]][i])
-      }
-            
-      # Handle settings with two levels
-    } else if (length(hierarchical_metadata[[i]]) == 2){
-      
-      #Create list if it does not exist
-      if (!is.list(potential_settings[[hierarchical_metadata[[i]][1]]])) { potential_settings[[hierarchical_metadata[[i]][1]]] = list() } #Need to make list if it doesnt exist since its two-level
-      
-      #Handle Nulls
-       if (is.null(parse_expr(metadata[[standard_low]][i]))) {
-         potential_settings[[hierarchical_metadata[[i]][1]]][hierarchical_metadata[[i]][2]] = list(parse_expr(metadata[[standard_low]][i]))
-         
-       } else {
-        potential_settings[[hierarchical_metadata[[i]][1]]][[hierarchical_metadata[[i]][2]]] = parse_expr( metadata[[standard_low]][i])
-      }
-            
+  for (row in hierarchical_metadata)  {
```

Comment:
We should probably go ahead and make this recursive so that it deals with N levels appropriately. Will be fairly similar to `getSettingValue()`

Reply:
filed #135